platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -342,24 +345,20 @@ public Response processSamlResponse(
 
   private void addSamlToSession(
       HttpServletRequest httpRequest, String realm, SecurityToken securityToken) {
-    if (securityToken == null) {
-      LOGGER.debug("Cannot add null security token to session.");
-      return;
-    }
-
-    HttpSession session = sessionFactory.getOrCreateSession(httpRequest);
-    Object sessionToken = getSecurityToken(session, realm);
-    if (sessionToken == null) {
-      addSecurityToken(session, realm, securityToken);
+    if (securityToken != null) {
+      HttpSession session = sessionFactory.getOrCreateSession(httpRequest);
+      Object sessionToken = getSecurityToken(session, realm);
+      if (sessionToken == null) {
+        addSecurityToken(session, realm, securityToken);
+        SecurityLogger.audit(
+            "Added SAML for user [{}] to session [{}]",
+            securityToken.getPrincipal().getName(),
+            Hashing.sha256().hashString(session.getId(), StandardCharsets.UTF_8).toString());
+        int minutes = 60;
+        session.setMaxInactiveInterval(minutes * 60);
+      }
     }
-    SecurityAssertion securityAssertion = new SecurityAssertionSaml(securityToken);
-    SecurityLogger.audit(
-        "Added SAML for user [{}] to session [{}]",
-        securityAssertion.getPrincipal().getName(),
-        Hashing.sha256().hashString(session.getId(), StandardCharsets.UTF_8).toString());
-    int minutes = 60;
-
-    session.setMaxInactiveInterval(minutes * 60);
+    LOGGER.debug("Problem adding security token to session.");
```

Comment:
📘 Refactored to:
1. Avoid calling `SecurityAssertion securityAssertion = new SecurityAssertionSaml(securityToken);`, which was making unit tests very difficult.
2. Only audit when the security token is successfully added.